### PR TITLE
Ignore errors when cleaning up remote access plugin

### LINKED ISSUE #1869: c8y-remote-access-plugin cleanup fails if support operation has already been deleted
**Describe the bug**

Purging `c8y-remote-access-plugin` fails if the `/etc/tedge/operations/c8y/c8y_RemoteAccessConnect` file no longer exists when the purging is executed. This block the removal of the package, and makes it very difficult for users to remove the package (as it requires the user to manually delete the `postrm` script before purging).

**Workaround**

Manually delete the `postrm` script prior to purging the package, or before calling the thin-edge.io uninstall script.

```sh
sudo rm -f /var/lib/dpkg/info/c8y-remote-access-plugin.prerm
curl -sSL https://raw.githubusercontent.com/thin-edge/thin-edge.io/main/uninstall-thin-edge_io.sh | sudo sh -s purge
```

**To Reproduce**

1. Install thin-edge (include `c8y-remote-access-plugin`)
    ```sh
    curl -fsSL https://raw.githubusercontent.com/thin-edge/thin-edge.io/main/get-thin-edge_io.sh | sudo sh -s
    ```

2. Manually remove the `c8y_RemoteAccessConnect` file

    ```sh
    sudo rm -rf /etc/tedge/operations/c8y/c8y_RemoteAccessConnect
    ```
    **note**
    * Technically this file might not be manually removed by a user, it is more likely that the `tedge-agent` which removes the `/etc/tedge/operations` folder on its post removal script `postrm`

4. Try to uninstall the package

    ```sh
    sudo apt-get purge c8y-remote-access-plugin
    ```

    Or alternatively you can try to purge it using the uninstall script
    
    ```sh
    curl -sSL https://raw.githubusercontent.com/thin-edge/thin-edge.io/main/uninstall-thin-edge_io.sh | sudo sh -s purge
    ```

Afterwards you may need to reset your setup you may need to delete the prerm file manually (because otherwise your package system may be broken).

```sh
sudo rm -f /var/lib/dpkg/info/c8y-remote-access-plugin.prerm
sudo apt-get purge c8y-remote-access-plugin
```

**Expected behavior**

* `c8y-remote-access-plugin` should not fail on removing a file or folder that does not exist (as the work has already been done, so the action should be idempotent)

**Screenshots**

The following log output shows the `c8y-remote-access-plugin` error log when trying to be purged via the install script.

```sh
$ curl -sSL https://raw.githubusercontent.com/thin-edge/thin-edge.io/main/uninstall-thin-edge_io.sh | sudo sh -s purge
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following packages will be REMOVED:
  c8y-remote-access-plugin*
0 upgraded, 0 newly installed, 1 to remove and 545 not upgraded.
1 not fully installed or removed.
After this operation, 2446 kB disk space will be freed.
(Reading database ... 75873 files and directories currently installed.)
Removing c8y-remote-access-plugin (0.10.0) ...
Error:
  × Removing supported operation at /etc/tedge/operations/c8y/
  │ c8y_RemoteAccessConnect
  ╰─▶ No such file or directory (os error 2)

dpkg: error processing package c8y-remote-access-plugin (--remove):
 installed c8y-remote-access-plugin package pre-removal script subprocess returned error exit status 1
dpkg: too many errors, stopping
Error:
  × Declaring supported operations
  ╰─▶ Creating the file failed: "/etc/tedge/operations/c8y/
      c8y_RemoteAccessConnect".

dpkg: error while cleaning up:
 installed c8y-remote-access-plugin package post-installation script subprocess returned error exit status 1
Errors were encountered while processing:
 c8y-remote-access-plugin
Processing was halted because there were too many errors.
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

**Environment (please complete the following information):**
 - OS [incl. version]: Debian 11
 - Hardware [incl. revision]: Raspberry Pi 3 (armv7)
 - System-Architecture [e.g. result of "uname -a"]: Linux raspberrypi 5.15.84-v7l+
 - thin-edge.io version [e.g. 0.1.0]: 0.10.0

**Additional context**

### PR DESCRIPTION
**Pull request by jarhodes314**
## Proposed changes

Fixes #1869 by ignoring cleanup errors.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue

#1869

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by didier-wenzek — APPROVED** [review]
Approved

**Comment by reubenmiller** [comment]
There are just some integration tests which I have already forwarded to @jarhodes314 which should be committed before we merge.

**Comment by reubenmiller** [comment]
Sorry @jarhodes314 my mistake. I’ll need to adjust the test to install the artifact from the current branch and not from main.

File: plugins/c8y_remote_access_plugin/src/main.rs [hunk 1]
```diff
@@ -174,4 +176,30 @@ mod tests {
...
+        // Verify that this doesn't panic
+        remove_supported_operation(
+            "/tmp/already-deleted-operations-via-removal-of-tedge-agent".into(),
+        );
+    }
+
+    fn create_example_operation(dir: &Utf8Path) -> Utf8PathBuf {
+        let operation_path = supported_operation_path(dir);
+        std::fs::create_dir_all(operation_path.parent().unwrap()).unwrap();
+        std::fs::File::create(&operation_path).unwrap();
```
Comment (albinsuresh) [inline plugins/c8y_remote_access_plugin/src/main.rs:201]: Just for future reference: we've got our own wrapper over `tempfile` called `TempTedgeDir` which makes creation of hierarchical files slightly easier  as follows:

```
TempTedgeDir::new()
            .dir("operations")
            .dir("c8y")
            .file("c8y_RemoteAccessConnect")
            .with_raw_content("content")
```

**Review by albinsuresh — APPROVED** [review]

**Pull request merged**